drv/transceivers-server/src/main.rs

```diff
@@ -112,6 +158,178 @@ impl ServerImpl {
     }
 }
 
+impl ServerImpl {
+    /// Returns the temperature from a CMIS transceiver.
+    ///
+    /// `port` is a logical port index, i.e. 0-31.
+    fn read_cmis_temperature(
+        &self,
+        port: LogicalPort,
+    ) -> Result<Celsius, FpgaError> {
+        const CMIS_TEMPERATURE_MSB: u8 = 14; // CMIS, Table 8-9
+        self.read_temperature_from_i16(port, CMIS_TEMPERATURE_MSB)
+    }
+
+    /// Returns the temperature from a SFF-8636 transceiver.
+    ///
+    /// `port` is a logical port index, i.e. 0-31.
+    fn read_sff8636_temperature(
+        &self,
+        port: LogicalPort,
+    ) -> Result<Celsius, FpgaError> {
+        const SFF8636_TEMPERATURE_MSB: u8 = 22; // SFF-8636, Table 6-7
+        self.read_temperature_from_i16(port, SFF8636_TEMPERATURE_MSB)
+    }
+
+    /// Trigger a read from the given port's given register, which is assumed to
+    /// be an `i16` containing 1/256 °C.
+    fn read_temperature_from_i16(
+        &self,
+        port: LogicalPort,
+        reg: u8,
+    ) -> Result<Celsius, FpgaError> {
+        self.transceivers.setup_i2c_read(reg, 2, port.as_mask())?;
+
+        #[derive(Copy, Clone, FromBytes, AsBytes)]
+        #[repr(C)]
```

Comment:
Does this also need `packed`? I think it's currently 4 bytes with a pad byte after `status`; not sure how that interacts with `zerocopy`.

Reply:
[`zerocopy::byteorder::I16`](https://docs.rs/zerocopy/latest/zerocopy/byteorder/struct.I16.html#) is unaligned, so this ends up being tightly packed!  I could switch to `repr(packed)` if you think it would be more clear, though.

Reply:
Ah, I didn't realize that! I should have, though - w.r.t. "not sure how [padding] interacts with zerocopy": it doesn't. If you were using regular `i16` here (which would have a pad byte) zerocopy would fail to derive `AsBytes`.

I don't think adding `repr(packed)` is necessary given all that.